src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.DragDrop.cs

```diff
@@ -262,7 +262,14 @@ private static void OnReorderCompleted(object sender, _DragEventArgs dragEventAr
 
 			that.m_tpPrimaryDraggedContainer = null;
 
-			that.ChangeSelectorItemsVisualState(true);
+			// CompleteReordering will remove the children and add them back on next measure.
+			// We defer ChangeSelectorItemsVisualState to the next measure so that the children are there and updated.
+			// An alternative could be to retrieve the children before CompleteReordering and then update the visual state here.
+#if HAS_UNO_WINUI
+			that.DispatcherQueue.TryEnqueue(Microsoft.UI.Dispatching.DispatcherQueuePriority.Low, () => that.ChangeSelectorItemsVisualState(true));
+#else
+			_ = that.Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Low, () => that.ChangeSelectorItemsVisualState(true));
+#endif
```

Comment:
Note: you can internally use DispatcherQueue on the UWP tree as well (just with a different namespace).

Reply:
https://github.com/unoplatform/uno/blob/d188320c82131a63f7c7044bad9625ce973212b7/src/Uno.UI/UI/Xaml/DependencyObject.cs#L16-L18